src/main/java/tr/com/srdc/cda2fhir/transform/ICDATransformer.java

```diff
@@ -21,35 +21,48 @@
  */
 
 import org.hl7.fhir.dstu3.model.Bundle;
+import org.hl7.fhir.dstu3.model.Identifier;
 import org.hl7.fhir.dstu3.model.Reference;
-import org.openhealthtools.mdht.uml.cda.ClinicalDocument;
 import org.openhealthtools.mdht.uml.cda.consol.ContinuityOfCareDocument;
 
 import tr.com.srdc.cda2fhir.util.IdGeneratorEnum;
 
 public interface ICDATransformer {
-    /**
-     * Returns a Reference for the patient of the CDA document
-     * @return A Reference that references the patient (i.e. recordTarget/patientRole) of the document
-     */
-    Reference getPatientRef();
+	/**
+	 * Returns a Reference for the patient of the CDA document
+	 * 
+	 * @return A Reference that references the patient (i.e.
+	 *         recordTarget/patientRole) of the document
+	 */
+	Reference getPatientRef();
 
-    /**
-     * A consistent unique resource id generator
-     * @return a unique resource id
-     */
-    String getUniqueId();
+	/**
+	 * A consistent unique resource id generator
+	 * 
+	 * @return a unique resource id
+	 */
+	String getUniqueId();
 
-    /**
-     * Sets the resource id generator format, which is either an incremental COUNTER or UUID
-     * @param idGen The id generator enumeration to be set
-     */
-    void setIdGenerator(IdGeneratorEnum idGen);
+	/**
+	 * Sets the resource id generator format, which is either an incremental COUNTER
+	 * or UUID
+	 * 
+	 * @param idGen The id generator enumeration to be set
+	 */
+	void setIdGenerator(IdGeneratorEnum idGen);
 
-    /**
-     * Transforms a Clinical Document Architecture (CDA) instance to a Bundle of corresponding FHIR resources
-     * @param cda A ClinicalDocument (CDA) instance to be transformed
-     * @return A FHIR Bundle that contains a Composition corresponding to the CDA document and all other resources that are referenced within the Composition.
-     */
-    Bundle transformDocument(ContinuityOfCareDocument cda);
+	/**
+	 * Transforms a Clinical Document Architecture (CDA) instance to a Bundle of
+	 * corresponding FHIR resources
+	 * 
+	 * @param cda             A ContinuityOfCareDocument (CDA) instance to be
+	 *                        transformed
+	 * @param encodedBody     The base64 encoded body from the original document
+	 * @param assemblerDevice An identifier with the name of the device doing the
+	 *                        assembling.
+	 * @return A FHIR Bundle that contains a Composition corresponding to the CDA
+	 *         document and all other resources that are referenced within the
+	 *         Composition.
+	 */
+	Bundle transformDocument(ContinuityOfCareDocument cda, String encodedBody, Identifier assemblerDevice);
```

Comment:
@mmccall I don't have an answer to this one? I think I changed it to `ContinuityOfCareDocument` due to some merge conflicts. Something about it working in Eclipse and not in the JAR? Anyway, I focused mostly on `encodedBody`. The `Identifier` is just to label Higgs.

Reply:
ContinuityOfCareDocument is there because we had to do it that way for the Java to run on the server (instead of CDA type), that's all fine. You should add some conditionals though around the assemblerDevice that if it isn't present it won't try to make the provenance object. Re: the encodedBody, the cleanest way would probably be not to pass it in, and just encode the ContinuityOfCareDocument object as base64 since that's what we passed it.

Reply:
I'm pretty sure CDAUtil has tools on it that would let you cast the document as a string, then encode that.